plaso/parsers/windefenderhistory.py

```diff
@@ -0,0 +1,429 @@
+      values_data_offset += data_size
+
+    return threat_tracking
+
+  def _ReadThreatTrackingHeader(self, threat_tracking_data):
+    """Reads the threat tracking header.
+
+      Args:
+          threat_tracking_data (bytes): threat tracking data.
+
+      Returns:
+          threat_tracking_header: threat tracking header.
+
+      Raises:
+          IOError: if the threat tracking header cannot be read.
+    """
+    data_type_map = self._GetDataTypeMap("threat_tracking_header")
+
+    threat_tracking_header = self._ReadStructureFromByteStream(
+        threat_tracking_data, 0, data_type_map
+    )
+
+    return threat_tracking_header
+
+  def _ReadThreatTrackingValue(self, threat_tracking_data, file_offset):
+    """Reads the threat tracking value.
+
+      Args:
+          threat_tracking_data (bytes): threat tracking data.
+          file_offset (int): offset of the threat tracking data relative to
+              the start of the file.
+
+      Returns:
+          tuple[threat_tracking_value, int]: threat tracking value and
+            data size.
+
+      Raises:
+          IOError: if the threat tracking value cannot be read.
+    """
+    data_type_map = self._GetDataTypeMap("threat_tracking_value")
+
+    context = dtfabric_data_maps.DataTypeMapContext()
+
+    threat_tracking_value = self._ReadStructureFromByteStream(
+        threat_tracking_data, file_offset, data_type_map, context=context
+    )
+
+    return threat_tracking_value, context.byte_size
+
+  def _CreateDateTime(self, date_time):
+    """Creates a date time value from the given string..
+      Args:
+          date_time (str): Date time string in nanoseconds since epoch.
+
+      Returns:
+          dfdatetime.TimeElements: date time object.
+    """
+    epoch = timedelta(microseconds=float(date_time / 10))
+    dt = datetime(1601, 1, 1) + epoch
+    te = dfdatetime_time_elements.TimeElements()
+    te.CopyFromDatetime(dt)
+    return te
+
+  def _ReadValue(self, file_object, file_offset):
+    """Reads the value.
+
+      Args:
+          file_object (file): file-like object.
+          file_offset (int): offset of the value relative to the
+            start of the file.
+
+      Returns:
+          object: value.
+
+      Raises:
+          IOError: if the value cannot be read.
+    """
+    data_type_map = self._GetDataTypeMap("detection_history_value")
+
+    value, _ = self._ReadStructureFromFileObject(
+        file_object, file_offset, data_type_map
+    )
+
+    value_object = None
+    if value.data_type in (0x00000000, 0x00000005, 0x00000006, 0x00000008):
+      value_object = value.value_integer
+    elif value.data_type == 0x0000000A:
+      value_object = value.value_filetime
+    elif value.data_type == 0x00000015:
+      value_object = value.value_string
+    elif value.data_type == 0x0000001E:
+      value_object = value.value_guid
+    elif value.data_type == 0x00000028:
+      value_object = value.data
```

Comment:
let's add a parser warning for unknown value data types

Reply:
Done